code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -552,3 +552,65 @@
 	qdel(M)
 	playsound(get_turf(N),'sound/items/ratchet.ogg',50,TRUE)
 	return
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/firefighter
+	name = "Ripley Heavy-Duty Frontier kit"
+	desc = "A complicated kit containing all the parts necessary to upgrade a MK-II's armour with additional armour plates and improved insulation around the electronics. Must be applied to a MK-II's frame and cannot be removed once applied. "
+	icon_state = "ripleyupgrade"
+
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/firefighter/can_attach(obj/mecha/working/ripley/M)
```

Comment:
I know you're probably porting this but, you need to replace all of these single letter variables with a meaningful name.

Reply:
yeah, it's just a copy-paste from the MKII upgrade that's already in the game (scroll up). Any idea how to inherit the code while changing the MKII stuff to point towards the MKIII mech?

Reply:
Right, changed all the repetitive vars excluding the 'E' vars to be more descriptive.

Reply:
going to close this, lemme know if the 'E's are a problem